modules/stac-server/opensearch_serverless.tf

```diff
@@ -0,0 +1,97 @@
+resource "aws_opensearchserverless_security_policy" "stac_server_opensearch_serverless_encryption_policy" {
+  count   = var.deploy_stac_opensearch_serverless ? 1 : 0
+  name    = lower(var.opensearch_stac_server_domain_name_override == null ? "${local.name_prefix}-stac-ep" : var.opensearch_stac_server_domain_name_override)
+  type    = "encryption"
+  policy  = jsonencode({
+    "Rules" = [
+      {
+        "Resource" = [
+          "collection/${lower(var.opensearch_stac_server_domain_name_override == null ? "${local.name_prefix}-stac-server" : var.opensearch_stac_server_domain_name_override)}"
+        ],
+        "ResourceType" = "collection"
+      }
+    ],
+    "AWSOwnedKey" = true
+  })
+}
+
+resource "aws_opensearchserverless_security_policy" "stac_server_opensearch_serverless_network_policy" {
+  count   = var.deploy_stac_opensearch_serverless ? 1 : 0
+  name    = lower(var.opensearch_stac_server_domain_name_override == null ? "${local.name_prefix}-stac-np" : var.opensearch_stac_server_domain_name_override)
+  type    = "network"
+  policy  = jsonencode([
+    {
+      Description = "Public access for collection endpoint",
+      Rules = [
+        {
+          ResourceType = "collection",
+          Resource = [
+            "collection/${lower(var.opensearch_stac_server_domain_name_override == null ? "${local.name_prefix}-stac-server" : var.opensearch_stac_server_domain_name_override)}"
+          ]
+        }
+      ],
+      AllowFromPublic = true,
+    },
+    {
+      Description = "Public access for dashboards",
+      Rules = [
+        {
+          ResourceType = "dashboard"
+          Resource = [
+            "collection/${lower(var.opensearch_stac_server_domain_name_override == null ? "${local.name_prefix}-stac-server" : var.opensearch_stac_server_domain_name_override)}"
+          ]
+        }
+      ],
+      AllowFromPublic = true
```

Comment:
should we use public or vpc here?

Reply:
Will be looking at this likely in the near future but not in our first iteration.